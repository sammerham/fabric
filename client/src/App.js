import { React, useState, useEffect } from 'react';
import MediaApi from './api/mediaApi';
import MediaList from './components/MediaList';

import {
  StyledButton,
  StyledMainTitle,
  StyledDivButton
} from './components/StyledComponents'
const KEY = 'localStorageMatrixKey';



/*
 * App
 *
 * This is the page where you rednder App
 * State: data

 * props: none
 * App ---->> MediaList ----> MediaItem
 */

function App() {
  const [data, setData] = useState([]);


  //! *********** Save and get data from local storage **************

  // get data from local storage in first render if any
  useEffect(() => {
    const SavedData = JSON.parse(localStorage.getItem(KEY));
    setData(data => SavedData);
  }, []);

  
  // save in local storage so you won't lose info on refresh
  useEffect(() => {
    localStorage.setItem(KEY, JSON.stringify(data));
  }, [data]);

//! ******************************************************************

// fn to handle click for button one
  const handleClickMatrix = () => {
    async function fetchMatrixData() {
      const MatrixData = await MediaApi.getAllMatrixMedia();
      
      setData(data => MatrixData.Search);
    }
    fetchMatrixData();
  };


// fn to handle click for button two
  const handleClickMatrixRevolutions = () => {
    async function fetchMatrixRevolutionsData() {
      const MatrixRevolutionsData = await MediaApi.getAllMatrixrevolutionsMedia();
      setData(d => MatrixRevolutionsData.Search);
    }
    fetchMatrixRevolutionsData();
  };


// fn to handle click for button three
  const handleClickMatrixReloaded = () => {
    async function fetchMatrixReloadedData() {
      const MatrixReloadedData = await MediaApi.getAllMatrixreloadedMedia();
      setData(dat => MatrixReloadedData.Search);
    }
    fetchMatrixReloadedData();
  };


  return (
    <div>
      <div > 
        <StyledMainTitle bg={'#448565'} className='App-sameh'>Media Information</StyledMainTitle>
        <StyledDivButton>
          <StyledButton bg={'#58755b'} onClick={handleClickMatrix}>Matrix</StyledButton>
          <StyledButton bg={'#58755b'} onClick={handleClickMatrixRevolutions}>Matrix Revolutions</StyledButton>
          <StyledButton bg={'#58755b'} onClick={handleClickMatrixReloaded}>Matrix Reloaded</StyledButton>
        </StyledDivButton>
      </div>
      <MediaList data={data}/>
    </div>
  );
}

export default App;
