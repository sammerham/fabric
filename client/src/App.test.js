import React from 'react'
import { screen, render, fireEvent, waitFor } from '@testing-library/react';
import App from './App';


// smoke test
test('renders without crashing', () => {
  render(<App />);
});

// snapShot
test("matches snapshot", function () {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});


test("checking if title and main buttons show on main app", function () {
  render(<App />);
  expect(screen.getByText('Media Information')).toBeInTheDocument()
  expect(screen.getByText('Matrix')).toBeInTheDocument()
  expect(screen.getByText('Matrix Revolutions')).toBeInTheDocument()
  expect(screen.getByText('Matrix Reloaded')).toBeInTheDocument()
});

test("checking matrix results show when clicking matrix button", async function () {
  render(<App />);
  const button = screen.getByText('Matrix');
  fireEvent.click(button)
  await screen.findByText('A Glitch in the Matrix')
  expect(screen.getByText('A Glitch in the Matrix')).toBeInTheDocument()
});

test("checking matrix revolutions results show when clicking matrix button", async function () {
  render(<App />);
  const button = screen.getByText('Matrix Revolutions');
  fireEvent.click(button)
  await screen.findByText('The Matrix Revolutions: Super Burly Brawl')
  expect(screen.getByText('The Matrix Revolutions: Super Burly Brawl')).toBeInTheDocument()
});


test("checking matrix reloaded results show when clicking matrix button", async function () {
  render(<App />);
  const button = screen.getByText('Matrix Reloaded');
  fireEvent.click(button)
  await screen.findByText('The Matrix Reloaded: The Exiles')
  expect(screen.getByText('The Matrix Reloaded: The Exiles')).toBeInTheDocument()
});