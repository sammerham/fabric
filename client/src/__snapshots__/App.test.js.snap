// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div>
    <div>
      <h1
        class="sc-iCfMLu cKMWxr App-sameh"
      >
        Media Information
      </h1>
      <div
        class="sc-gKclnd hrMEC"
      >
        <button
          class="sc-furwcr grhNup"
        >
          Matrix
        </button>
        <button
          class="sc-furwcr grhNup"
        >
          Matrix Revolutions
        </button>
        <button
          class="sc-furwcr grhNup"
        >
          Matrix Reloaded
        </button>
      </div>
    </div>
    <div />
  </div>
</div>
`;
